sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillableColumnarBatch.scala

```diff
@@ -100,27 +101,128 @@ class SpillableColumnarBatchImpl (
    */
   override def numRows(): Int = rowCount
 
-  private def withRapidsBuffer[T](fn: RapidsBuffer => T): T = {
-    withResource(RapidsBufferCatalog.acquireBuffer(handle)) { rapidsBuffer =>
-      fn(rapidsBuffer)
+  override lazy val sizeInBytes: Long = handle.sizeInBytes
+
+  /**
+   * Set a new spill priority.
+   */
+  override def setSpillPriority(priority: Long): Unit = {
+    // TODO: handle.setSpillPriority(priority)
+  }
+
+  override def getColumnarBatch(): ColumnarBatch = {
+    GpuSemaphore.acquireIfNecessary(TaskContext.get())
+    handle.materialize(sparkTypes)
+  }
+
+  override def incRefCount(): SpillableColumnarBatch = {
+    if (refCount <= 0) {
+      throw new IllegalStateException("Use after free on SpillableColumnarBatchImpl")
     }
+    refCount += 1
+    this
   }
 
-  override lazy val sizeInBytes: Long =
-    withRapidsBuffer(_.memoryUsedBytes)
+  /**
+   * Remove the `ColumnarBatch` from the cache.
+   */
+  override def close(): Unit = {
+    refCount -= 1
+    if (refCount == 0) {
+      // closing my reference
+      handle.close()
+    }
+    // TODO this is causing problems so we need to look into this
```

Comment:
Should we just remove this then? It feels like that should be a separate issue as this is already a large enough PR. (like 10161)

Reply:
Okay Sorry it looks like these comments were here already. This is just copied from previous code.